verilog/analysis/verilog_linter.cc

```diff
@@ -77,12 +83,226 @@ using verible::LintWaiver;
+    }
+  }
+
+  for (auto violation : violations) {
+    HandleViolation(*violation.violation, base, path, violation.status->url,
+                    violation.status->lint_rule_name);
+  }
+}
+
+void ViolationPrinter::HandleLintRuleStatuses(
+    const std::vector<verible::LintRuleStatus>& statuses,
+    absl::string_view base, absl::string_view path) {
+  verible::LintStatusFormatter formatter(base);
+
+  formatter_ = &formatter;
+  LintStatusHandler::HandleLintRuleStatuses(statuses, base, path);
+  formatter_ = nullptr;
+}
+
+void ViolationPrinter::HandleViolation(const verible::LintViolation& violation,
+                                       absl::string_view base,
+                                       absl::string_view path,
+                                       absl::string_view url,
+                                       absl::string_view rule_name) {
+  formatter_->FormatViolation(stream_, violation, base, path, url, rule_name);
+  (*stream_) << std::endl;
+}
+
+static void PrintFix(std::ostream& stream, absl::string_view text,
+                     const verible::AutoFix& fix) {
+  std::string after = fix.Apply(text);
+  verible::LineDiffs diff(text, after);
+
+  verible::LineDiffsToUnifiedDiff(stream, diff, 1);
+}
+
+void ViolationFixer::CommitFixes(absl::string_view source_content,
+                                 absl::string_view source_path,
+                                 const verible::AutoFix& fix) const {
+  if (fix.Edits().empty()) {
+    return;
+  }
+  std::string fixed_content = fix.Apply(source_content);
+
+  if (patch_stream_) {
+    verible::LineDiffs diff(source_content, fixed_content);
+    verible::LineDiffsToUnifiedDiff(*patch_stream_, diff, 1, source_path);
+  } else {
+    const absl::Status write_status =
+        verible::file::SetContents(source_path, fixed_content);
+    if (!write_status.ok()) {
+      LOG(ERROR) << "Failed to write fixes to file '" << source_path
+                 << "': " << write_status.ToString();
+      return;
+    }
+  }
+}
+
+void ViolationFixer::HandleLintRuleStatuses(
+    const std::vector<LintRuleStatus>& statuses, absl::string_view base,
+    absl::string_view path) {
+  verible::AutoFix fix;
+
+  fix_ = &fix;
+  ViolationPrinter::HandleLintRuleStatuses(statuses, base, path);
+  fix_ = nullptr;
+
+  CommitFixes(base, path, fix);
+}
+
+void ViolationFixer::HandleViolation(const verible::LintViolation& violation,
+                                     absl::string_view base,
+                                     absl::string_view path,
+                                     absl::string_view url,
+                                     absl::string_view rule_name) {
+  std::stringstream violation_message;
+  formatter_->FormatViolation(&violation_message, violation, base, path, url,
+                              rule_name);
+  (*stream_) << violation_message.str() << std::endl;
+
+  if (violation.autofixes.size() > 0) {
```

Comment:
instead of comparing size() with 0, just use a call to `empty()`
(reason: we know that `empty()` is probably a O(1) operation, while size() might not be for all containers. So it is a good habit to do)

Reply:
Done